tasks/image_size.js

```diff
@@ -7,53 +7,92 @@
 
 'use strict';
 
+var path = require('path');
+var chalk = require('chalk');
+var sizer = require('image-size');
+
 module.exports = function(grunt) {
-	var path = require('path');
-	var chalk = require('chalk');
-	var sizer = require('image-size');
-	var types = [
-		'.png',
-		'.gif',
-		'.bmp',
-		'.psd',
-		'.jpg',
-		'.tiff',
-		'.webp'
-	];
-
-	grunt.registerMultiTask('image_size', 'retrieve image saze information', function() {
-		var options = this.options({
-			configObject : ''
-		});
-
-		this.files.forEach(function (file) {
-			if ((!options.configObject || !options.configObject.length) && !file.dest) {
-				grunt.fail.warn('Neither options.configObject nor dest specified for this task.');
-				return;
-			}
-
-			var output = [];
-			file.src.forEach(function (src) {
-				if (grunt.file.exists(src) && grunt.file.isFile(src) && types.indexOf(path.extname(src).toLowerCase()) !== -1) {
-					var dimensions = sizer(src);
-					output.push({
-						name : src,
-						width : dimensions.width,
-						height : dimensions.height
-					});
-
-					grunt.verbose.writeln('dimensions of ' + src + ' width: ' + dimensions.width.toString() + ', height: ' + dimensions.height.toString());
-				}
-			});
-
-			if (options.configObject && options.configObject.length) {
-				grunt.config.set(options.configObject, output);
-			}
-
-			if (file.dest) {
-				grunt.file.write(file.dest, JSON.stringify(output));
-				grunt.log.writeln('File ' + chalk.cyan(file.dest) + ' created.');
-			}
-		});
-	});
+  grunt.registerMultiTask(
+    'image_size',
+    'Retrieve images size information',
+    function() {
+      var options = this.options({
+        processName: undefined,
+        processEntry: undefined,
+        processSizes: undefined,
+        configObject: undefined,
+        replacer: null,
+        space: 2
+      });
+
+      if (!this.files.length) {
+        return grunt.log.error('No files or `options.configObject` specified.');
+      }
+
+      var processedFiles = 0;
+
+      this.files.forEach(function(file) {
+        if (!file.src.length)
+          return grunt.log.error(
+            'No source files specified for ' + chalk.cyan(dest) + '.'
+          );
+
+        if (!file.dest) return grunt.log.error('No dest file specified.');
```

Comment:
- return Log object again
- if configObject is specified, there is no error here
- if both configObject and file.dest undefined - that is incorrect config (and you need to check it here and report with grunt.fail.warn)

Reply:
Yeap, missed that. Fixed in related to previous comment commit.

It will report this: https://i.imgur.com/A7ZsQK3.png